packages/commonwealth/client/scripts/views/modals/AuthModal/AuthModal.tsx

```diff
@@ -0,0 +1,160 @@
+import { ChainBase, WalletSsoSource } from '@hicommonwealth/core';
+import useWallets from 'client/scripts/hooks/useWallets';
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import app from 'state';
+import AuthButton from '../../components/AuthButton';
+import { AuthTypes } from '../../components/AuthButton/types';
+import { CWIcon } from '../../components/component_kit/cw_icons/cw_icon';
+import { CWText } from '../../components/component_kit/cw_text';
+import { CWModal } from '../../components/component_kit/new_designs/CWModal';
+import {
+  CWTab,
+  CWTabsRow,
+} from '../../components/component_kit/new_designs/CWTabs';
+import './AuthModal.scss';
+
+type AuthModalProps = {
+  isOpen: boolean;
+  onClose: () => any;
+  onSuccess?: () => any;
+  showWalletsFor?: ChainBase.CosmosSDK | ChainBase.Ethereum;
+};
+
+const AuthModal = ({
+  isOpen,
+  onClose,
+  onSuccess,
+  showWalletsFor,
+}: AuthModalProps) => {
+  const [activeTabIndex, setActiveTabIndex] = useState<number>(0);
+
+  const { wallets, onWalletSelect, onSocialLogin } = useWallets({
+    onModalClose: onClose,
+    onSuccess,
+  });
+
+  const tabsList = [
+    {
+      name: 'Wallet',
+      options: [
+        // Note: If `showWalletsFor` is present then show wallets for that chain, else
+        // only show ethereum based wallets options on non-community pages or on ethereum based communities
+        ...((!showWalletsFor
+          ? !app?.chain?.base
+          : showWalletsFor === ChainBase.Ethereum) ||
+        app?.chain?.base === ChainBase.Ethereum
+          ? ['walletConnect']
+          : []),
+        // only show cosmos based wallets on non-community pages or on cosmos based communities
+        ...(wallets || [])
+          .filter(
+            (wallet) =>
+              ((!showWalletsFor
+                ? !app?.chain?.base
+                : showWalletsFor === ChainBase.CosmosSDK) ||
+                app?.chain?.base === ChainBase.CosmosSDK) &&
+              wallet.chain === ChainBase.CosmosSDK,
+          )
+          .map((wallet) => wallet.name),
+        // only show solana based wallets on solana based communities
+        ...(wallets || [])
+          .filter(
+            (wallet) =>
+              app?.chain?.base === ChainBase.Solana &&
+              wallet.chain === ChainBase.Solana,
+          )
+          .map((wallet) => wallet.name),
+        // only show substrate based wallets on substrate based communities
+        ...(wallets || [])
+          .filter(
+            (wallet) =>
+              app?.chain?.base === ChainBase.Substrate &&
+              wallet.chain === ChainBase.Substrate,
+          )
+          .map((wallet) => wallet.name),
```

Comment:
despite nice comments, it is pretty hard to read it, especially those multilines and nested ternary operators
1. How about abstract out filter function so you would pass only `ChainBase` and get filtered wallets
2. let's try to use more variables and then build up on top of them, maybe something like 
const cosmosWallets = []; const substrateWallets = [];

Reply:
done.